Source/WebKit/UIProcess/mac/WebPageProxyMac.mm

```diff
@@ -589,18 +589,37 @@ static inline bool expectsLegacyImplicitRubberBandControl()
     [nsMenu setAllowsContextMenuPlugIns:false];
     for (unsigned i = 0; i < contextMenu.items.size(); i++) {
         auto& item = contextMenu.items[i];
-        
+        auto isOpenWithDefaultViewerItem = item.action == WebCore::ContextMenuItemTagOpenWithDefaultApplication;
+
         if (item.separator == ContextMenuItemIsSeparator::Yes) {
             [nsMenu insertItem:[NSMenuItem separatorItem] atIndex:i];
             continue;
         }
-        
+
         RetainPtr nsItem = adoptNS([[NSMenuItem alloc] init]);
-        [nsItem setTitle:item.title.createNSString().get()];
+
+#if ENABLE(CONTEXT_MENU_IMAGES_FOR_INTERNAL_CLIENTS)
+        auto shouldSetImage = m_preferences->contextMenuImagesForInternalClientsEnabled();
+#endif
```

Comment:
Nit: If you define `shouldSetImage` like this, you won't need to use the platform guard in multiple places. Optional though.

```suggestion
    auto shouldSetImage = [preferences = m_preferences] {
#if ENABLE(CONTEXT_MENU_IMAGES_FOR_INTERNAL_CLIENTS)
        return preferences->contextMenuImagesForInternalClientsEnabled();
#else
        return false;
#endif
    }();
#endif
```